app/models/geocode_logic.rb

```diff
@@ -0,0 +1,118 @@
+# frozen_string_literal: true
+
+class GeocodeLogic
+  def geocode_fields(result, address, address1, address2)
+    geocoded = Geocoder.coordinates(address)
+    geocoded1 = Geocoder.coordinates(address1)
+    geocoded2 = Geocoder.coordinates(address2)
+    geocoded3 = Geocoder.coordinates("#{result.city}, #{result.state}, #{result.zip}")
+    if geocoded.present?
+      update_mismatch(result, geocoded)
+    elsif geocoded1.present?
+      update_mismatch(result, geocoded1)
+    elsif geocoded2.present?
+      update_mismatch(result, geocoded2)
+    else
+      check_bad_address(result, geocoded3)
+    end
+  end
```

Comment:
```suggestion
  def geocode_fields(result, addresses)
    geocoded_result = Geocoder.coordinates("#{result.city}, #{result.state}, #{result.zip}")
    addresses.each do |address|
      geocoded = Geocoder.coordinates(address)
      if geocoded.present?
        update_mismatch(result, geocoded)
      else
        check_bad_address(result, geocoded_result)
      end
    end
  end
```

Reply:
Passing in the addresses as an array could help shorten and clean up this method.